source/clear-linux/tutorials/kubernetes.rst

```diff
@@ -0,0 +1,303 @@
+      CRI-O + *kata-runtime*:
+
+      #. Enable crio.service:
+
+         .. code-block:: bash
+
+            sudo systemctl enable crio.service
+
+      #. Restart services:
+
+         .. code-block:: bash
+
+            sudo systemctl restart crio
+            sudo systemctl restart kubelet
+
+Run Kubernetes for the first time
+*********************************
+
+#. Prepare your system to run Kubernetes for the first time with the
+   following commands:
+
+   .. code-block:: bash
+
+      sudo systemctl daemon-reload
+      sudo systemctl restart crio
+      sudo systemctl restart docker
+
+#. Initialize the master control plane with the following command, either:
+
+   a) If you are running Kubernetes with Docker + runc:
+
+      .. code-block:: bash
+
+         sudo -E kubeadm init --pod-network-cidr 10.244.0.0/16 --ignore-preflight-errors=SystemVerification
+
+   or:
+
+   b) If you are running Kubernetes with CRI-O + kata-runtime:
+
+      .. code-block:: bash
+
+         sudo -E kubeadm init --pod-network-cidr 10.244.0.0/16 --cri-socket=/run/crio/crio.sock
+
+Once your master control is successfully initialized, instructions on how to
+use your cluster and its *IP*, *token*, and *hash* values are displayed. It is
+important that you note these cluster values because they will be needed when
+joining worker nodes to the cluster and some of them have a valid period. The
+values are presented in a format similar to:
+
+.. code-block:: bash
+
+   kubeadm join <master-ip>:<master-port> --token <token> --discovery-token-ca-cert-hash <hash>
+
+
+**Congratulations!**
+
+You've successfully installed and set up Kubernetes in |CL| using Docker and
+runc or CRI-O and kata-runtime. You are now ready to follow on-screen
+instructions to deploy a pod network to the cluster and join worker nodes with
+the displayed token and IP information.
+
+Related Topics
+**************
+
+Read the Kubernetes documentation to learn more about: 
+
+* `Understanding basic Kubernetes architecture`_
+
+* `Deploying an application to your cluster`_
+
+* `Installing a pod network add-on`_
+
+* `Joining your nodes`_
+
+Package configuration customization in Clear Linux (Optional)
+*************************************************************
+
+|CL| is a stateless system that looks for user-defined package configuration
+files in the :file:`/etc/<package-name>` directory to be used as default. If
+user-defined files are not found, |CL| uses the distribution-provided
+configuration files for each package.
+
+If you customize any of the default package configuration files, you **must**
+store the customized files in the :file:`/etc/` directory. If you edit any of
+the distribution-provided default files, your changes will be lost in the next
+system update.
+
+For example, to customize CRI-O configuration in your system you can run the
+following commands:
+
+.. code-block:: bash
+
+   sudo mkdir /etc/crio
+   sudo cp /usr/share/defaults/crio/crio.conf /etc/crio/
```

Comment:
this is potentially buggy. remember, ```kata-runtime``` installs ```/etc/crio/crio.conf``` and here we're overwriting those changes.